### PR TITLE
Fix Macro-F1 & Use train labels by default

### PR DESCRIPTION
**Pull request by Eleven1Liu**
**Description** 
1. Use train labels by default. If `include_test_labels` is true, use labels in both training and test dataset.
2. Add class `Macro-F1` for calculating the mean f1 scores of ALL labels in the dataset like `torchmetrics==0.3.2` does.

### CODE REVIEW

File: libmultilabel/nn/data_utils.py [hunk 1]
```diff
@@ -227,27 +227,32 @@ def load_or_build_text_dict(
...
     if label_file:
-        logging.info('Load labels from {label_file}')
+        logging.info(f'Load labels from {label_file}.')
         with open(label_file, 'r') as fp:
             classes = sorted([s.strip() for s in fp.readlines()])
     else:
+
+        splits = datasets.keys() if include_test_labels else datasets.keys() - {'test'}
         classes = set()
-        for dataset in datasets.values():
```
Comment (Eleven1Liu) [inline libmultilabel/nn/data_utils.py:247]:
```suggestion
       for k, v in dataset.items():
           ..
```

File: libmultilabel/metrics.py [hunk 2]
```diff
@@ -40,12 +40,57 @@ def compute(self):
...
+    def compute(self):
+        label_prec = torch.nan_to_num(self.tp_sum / self.preds_sum, posinf=0.)
+        label_recall = torch.nan_to_num(self.tp_sum / self.target_sum, posinf=0.)
+
+        if self.another_macro_f1:
+            macro_prec = torch.mean(label_prec)
+            macro_recall = torch.mean(label_recall)
+            return 2 * (macro_prec * macro_recall) / (macro_prec + macro_recall + 1e-10)
+        else:
+            label_f1 = 2 * (label_prec * label_recall) / (label_prec + label_recall + 1e-10)
```
Comment (cjlin1) [inline libmultilabel/metrics.py:84]: I think we can use 2*tp_sum/(preds_sum + target_sum) here
then lines 76-77 can be merged to 80-81

**Review by cjlin1 — APPROVED** [review]
